api/base/exceptions.py

```diff
@@ -80,20 +88,87 @@ class Conflict(APIException):
     default_detail = ('Resource identifier does not match server endpoint.')
 
 
-class InvalidQueryStringError(JSONAPIException):
+class JSONAPIParameterException(JSONAPIException):
+    def __init__(self, detail=None, parameter=None):
+        source = {
+            'parameter': parameter
+        }
+        super(JSONAPIParameterException, self).__init__(detail=detail, source=source)
+
+
+class JSONAPIAttributeException(JSONAPIException):
+    def __init__(self, detail=None, attribute=None):
+        source = {
+            'pointer': '/data/attributes/{}'.format(attribute)
+        }
+        super(JSONAPIAttributeException, self).__init__(detail=detail, source=source)
+
+
+class InvalidQueryStringError(JSONAPIParameterException):
     """Raised when client passes an invalid value to a query string parameter."""
     default_detail = 'Query string contains an invalid value.'
     status_code = http.BAD_REQUEST
 
-    def __init__(self, detail=None, parameter=None):
-        super(InvalidQueryStringError, self).__init__(detail=detail, source={
-            'parameter': parameter
-        })
 
+class InvalidFilterOperator(JSONAPIParameterException):
+    """Raised when client passes an invalid operator to a query param filter."""
+    status_code = http.BAD_REQUEST
+
+    def __init__(self, detail=None, value=None, valid_operators=None):
```

Comment:
Currently what we do is (approximately) this:
```
def __init__(self, detail=None, value=None, valid_operators=None):
    valid_operators = valid_operators or ['eq', 'lt', 'lte', 'gt', 'gte', 'contains', 'icontains']
     valid_operators = ', '.join(valid_operators)
```

The main advantage of moving that first statement into the call signature is dynamically generated help; the call signature is less accurate than it needs to be for this case.
```
def __init__(self, detail=None, value=None, 
    valid_operators=('eq', 'lt', 'lte', 'gt', 'gte', 'contains', 'icontains')):
```